crates/bevy_reflect/src/impls/smallvec.rs

```diff
@@ -143,43 +131,16 @@ where
     }
 }
 
-impl<T: SmallArray + TypePath + Send + Sync> Reflect for SmallVec<T>
-where
-    T::Item: FromReflect + MaybeTyped + TypePath,
-{
-    fn into_any(self: Box<Self>) -> Box<dyn Any> {
-        self
-    }
-
-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
-    fn as_any_mut(&mut self) -> &mut dyn Any {
-        self
-    }
-
-    fn into_reflect(self: Box<Self>) -> Box<dyn Reflect> {
-        self
-    }
-
-    fn as_reflect(&self) -> &dyn Reflect {
-        self
-    }
-
-    fn as_reflect_mut(&mut self) -> &mut dyn Reflect {
-        self
-    }
-
-    fn set(&mut self, value: Box<dyn Reflect>) -> Result<(), Box<dyn Reflect>> {
-        *self = value.take()?;
-        Ok(())
-    }
-}
+impl_full_reflect!(
+    <T> for SmallVec<T>
+    where
+        T: SmallArray + TypePath + Send + Sync,
+        T::Item: FromReflect + Reflect + MaybeTyped + TypePath
```

Comment:
this means we won't support `SmallVec<Box<dyn Reflect>>` (as `Box<dyn Reflect>` isn't `FromReflect`) ?

Reply:
Yep! The Future Work section goes into a bit more detail regarding this, but it also means we don't yet support types like `Vec<Box<dyn Reflect>>`, etc.

The goal is to address that limitation in a followup PR

Reply:
Ok, so for now lets get this in a followup!